colossalai/inference/config.py

```diff
@@ -289,6 +291,15 @@ def to_generation_config(self, model_config) -> GenerationConfig:
 
         return GenerationConfig.from_dict(meta_config)
 
+    def to_model_inference_config(self) -> "ModelInferenceConfig":
```

Comment:
Where is it going to be used

Reply:
This method together with `ModelInferenceConfig` class is primarily designed as a wrapper of params, including `dtype`, `use_cuda_kernel`, `use_flash_attn`, `use_spec_dec`, which determines what ops and generation strategies that `InferenceEngine` should use for inference. These params configure modellings, e.g. `NopaddingLlama` during runtime, so it is better to configure them (as a ModelInferenceConfig object) when initializing the model, rather than in the model forward pass.

Since currently we are carefully considering if the refactor should touch the `shardformer` interfaces, I will delete this method and `ModelInferenceConfig` for this PR.